client/src/reducers/index.js

```diff
@@ -27,14 +27,14 @@ const appReducer = combineReducers({
 // Any update that need multiple branchs of the store tree go here
 const rootReducer = (state, action) => {
   switch (action.type) {
-    case types.SYNC_CALENDAR_DAY:
+    case types.ADD_CALENDAR_DAY:
       let selectedDay = action.selectedDay;
       let selectedMeal = action.selectedMeal;
       let selectedWeek = state.planner.selectedWeek;
-      let updatedState = _.extend({}, state);
+      let updatedState = JSON.parse(JSON.stringify(state));
```

Comment:
It's making a copy of it.  React wasn't noticing the certain components were updating.  It's a somewhat hack-y solution I admit